components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -379,27 +383,36 @@ handler.on(PanelActions.signTransactionHardware.type, async (store, messageData:
   }
 
   await navigateToConnectHardwareWallet(store)
-  const info = hardwareAccount.hardware
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-  const signed = await signRawTransactionWithHardwareKeyring(info.vendor as HardwareVendor, info.path, messageData.rawMessage, messageData.coin, () => {
-    store.dispatch(PanelActions.signTransaction([messageData]))
-  })
+  const coin = messageData.account.accountId.coin
+  const info = messageData.account.hardware
+  assert(info)
+
+  const signed = await signRawTransactionWithHardwareKeyring(
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    info.vendor as HardwareVendor,
```

Comment:
do we need this type assertion?

Reply:
removed
https://github.com/brave/brave-core/pull/19586/commits/de8a47305ae846acfbfed7247e3d7b5a88096ad7